### PR TITLE
Update testing.md

### PR DESCRIPTION
**Pull request by andyblundell**
This is a set of changes intended to make expectations clearer in terms of test activities/practices within teams, including:

1. Peer review consideration around tests / testing
2. Principle of build-for-testability
3. Suggestions around coaching / mentoring within teams in terms of testing

### CODE REVIEW

File: practices/testing.md [hunk 1]
```diff
@@ -26,6 +26,7 @@
 - **Quality is the whole team's responsibility**
   - Education on testing and testing principles should be important to the whole team.
   - Quality approaches should be driven as a team and implemented by everyone.
+  - Teams should consider running regular coaching / mentoring sessions to support colleagues who are less experienced in testing to grow their skills, for example by no-blame group discussions to identify edge-case tests which have so far been missed, tests positioned incorrectly in the [test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html), etc.
```
Comment (regularfry) [inline practices/testing.md:29]:
```suggestion
  - Teams should consider running regular coaching / mentoring sessions to support colleagues who are less experienced in testing to grow their skills, for example by no-blame group discussions to identify edge-case tests which have so far been missed, tests positioned incorrectly in the [test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html), and pairing with a tester navigating so that the driver learns the necessary skills.
```

File: patterns/everything-as-code.md [hunk 2]
```diff
@@ -70,6 +70,9 @@ While effective testing is the best way to detect bugs or non-functional problem
 - Is the code clear and simple?
 - Is the code layout and structure consistent with agreed style and other code? (please see [enforce code formatting](enforce-code-formatting.md))
 - Would it easily allow future modification to meet slightly different needs, e.g. ten times the required data size or throughput?
+- Is it [built for testability](../practices/structured-code.md)?
+- Are the automated tests positioned appropriately in the [test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html), triggered appropriately in CI builds, etc?
```
Comment (regularfry) [inline patterns/everything-as-code.md:74]:
```suggestion
- Are the automated tests positioned appropriately in the [test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html), triggered appropriately in CI builds, and do they block the build when they fail?
```

File: insights/review.md [hunk 3]
```diff
@@ -167,7 +167,10 @@ You may wish to score each individual component or system separately for these a
 #### 9. Testing
 
 - We have great test coverage.
-- Testing is everyone's responsibility.
+- Testing is everyone's responsibility and test is a first-class concern.
```
Comment (regularfry) [inline insights/review.md:170]:
```suggestion
- Testing is everyone's responsibility and test is a first-class concern.
- A failing test suite in CI gets immediate attention.
```

**Review by regularfry — CHANGES_REQUESTED** [review]

**Review by regularfry — APPROVED** [review]

**Review by regularfry — APPROVED** [review]
